src/status_im/contexts/wallet/events.cljs

```diff
@@ -202,6 +202,32 @@
                                     (first derived-address-details)]))]
      {:fx [[:dispatch [:wallet/create-derived-addresses account-details on-success]]]})))
 
+(defn finalize-new-keypair
+  [_ [{:keys [sha3-pwd new-keypair]}]]
+  {:fx [[:json-rpc/call
+         [{:method     "accounts_addKeypair"
+           :params     [sha3-pwd new-keypair]
+           :on-success [:wallet/add-account-success (string/lower-case (:address new-keypair))]
+           :on-error   #(log/info "failed to create keypair " %)}]]]})
+
+(rf/reg-event-fx :wallet/finalize-new-keypair finalize-new-keypair)
+
+(defn get-keypairs
+  [_]
+  {:fx [[:json-rpc/call
+         [{:method     "accounts_getKeypairs"
+           :params     []
+           :on-success [:wallet/get-keypairs-success]
+           :on-error   #(log/info "failed to get keypairs " %)}]]]})
+
+(rf/reg-event-fx :wallet/get-keypairs get-keypairs)
+
+(defn get-keypairs-success
+  [{:keys [db]} [keypairs]]
+    {:db (assoc-in db [:wallet :keypairs] keypairs)})
```

Comment:
@OmarBasem - It's better we sanitize the data (renaming of keys, clearing nil, and setting defaults ..etc) before we use it in the UI.